include/dlaf/matrix/copy_tile.h

```diff
@@ -89,5 +122,63 @@ void copy(TileElementSize region, TileElementIndex in_idx, const matrix::Tile<co
   dlaf::tile::lacpy<T>(region, in_idx, in, out_idx, out);
 }
 
+#define DLAF_MAKE_FUNCTION_OBJECT(fname)     \
+  constexpr struct fname##_t {               \
+    template <typename... Ts>                \
+    decltype(auto) operator()(Ts&&... ts) {  \
+      return fname(std::forward<Ts>(ts)...); \
+    }                                        \
+  } fname##_o {                              \
+  }
+
+DLAF_MAKE_FUNCTION_OBJECT(copy);
+
+namespace internal {
+template <typename T>
+struct UnwrapFuture {
+  template <typename U>
+  static decltype(auto) call(U&& u) {
+    return std::forward<U>(u);
+  }
+};
+
+template <typename T, Device D>
+struct UnwrapFuture<hpx::future<Tile<T, D>>> {
+  template <typename U>
+  static auto call(U&& u) {
+    auto t = u.get();
+    return t;
+  }
+};
+
+template <typename T>
+struct dummy;
+
+template <typename F>
+struct UnwrapExtendTiles {
+  F f;
+
+  template <typename... Ts>
+  auto operator()(Ts&&... ts) {
+    // Extract values from futures (not shared_futures)
+    auto t = hpx::make_tuple<>(UnwrapFuture<std::decay_t<Ts>>::call(std::forward<Ts>(ts))...);
+
+    // Call f with all futures (not just future<tile>) unwrapped.
+    hpx::invoke_fused(hpx::util::unwrapping(f), t);
+
+    // Finally, we extend the lifetime of read-write tiles directly and
+    // read-only tiles wrapped in shared_futures by returning them here in a
+    // tuple
+    // TODO: *Only* return tiles, no other parameters?
+    return t;
+  }
```

Comment:
This is lots of unwrapping... I still need to see if this impacts compile-times in any way.

Reply:
@albestro FYI this is the "magic" (combined with the CUDA/cuBLAS executors) that extends the tile lifetimes.